### PR TITLE
Upgrade to Node >= 8

### PR DESCRIPTION
**Pull request by fcastilloec**
Requires https://github.com/electron-userland/electron-installer-common/pull/27 to be released.
Closes #276 as this PR is built on top of it.

### CODE REVIEW

File: test/helpers/describe_cli.js [hunk 1]
```diff
@@ -40,22 +38,23 @@ module.exports = function (desc, asar, options) {
...
-          .then(() => {
-            throw new Error('delta `.nupkg` was created')
-          }).catch(error => chai.expect(error.message).to.have.string('no such file or directory'))
+      it('does not generate a delta `.nupkg` package', async () => {
+        try {
+          await testAccess(`${options.dest}/${appName}-0.0.1-delta.nupkg'`)
+          throw new Error('delta `.nupkg` was created')
+        } catch (error) {
+          chai.expect(error.message).to.have.string('no such file or directory')
+        }
```
Comment (malept) [inline test/helpers/describe_cli.js]: Probably a better way to write these large `testAccess` blocks:

```javascript
chai.expect(await fs.pathExists(`${options.dest}/${appName}-0.0.1-delta.nupkg'`)).to.be.false
```

File: package.json [hunk 2]
```diff
@@ -26,32 +26,30 @@
...
+    "node": ">=8.0.0"
   },
   "dependencies": {
     "debug": "^4.1.0",
-    "electron-installer-common": "^0.6.1",
-    "fs-extra": "^7.0.1",
-    "glob": "^7.1.3",
-    "glob-promise": "^3.4.0",
+    "electron-installer-common": "^0.7.0",
+    "fs-extra": "^8.0.1",
```
[the remark this answers is not in the document]
Comment (fcastilloec) [reply package.json]: We only use `copy` and `ensureDir` from `fs-extra` so maybe that's why. The good thing is that all test are passing

File: .travis.yml [hunk 3]
```diff
@@ -7,15 +7,17 @@ addons:
       - wine
 
 env:
+  - DEBUG="electron-installer-windows" NODE_VERSION="12"
   - DEBUG="electron-installer-windows" NODE_VERSION="10"
```
Comment (malept) [inline .travis.yml]: In the interests of CI resource consumption, I wonder if it's worth just testing against the oldest and newest supported LTS versions.

Comment (fcastilloec) [reply .travis.yml]: I don't think it's a bad idea but I'm not sure how I feel about it. Especially for Appveyor, which ends up testing for each version, both x86, and x64.
Why not, I'll remove node 10 from testing, and use only 8 and 12. We should do the same in all other modules.

Comment (malept) [reply .travis.yml]: > We should do the same in all other modules.

Yep, I was planning on it unless you had any objections.

Comment (fcastilloec) [reply .travis.yml]: Not objections. I always had a feeling that we should test for each version we're using but when thinking carefully about it, there's really no good reason (that I can think of) of just using the oldest and newest LTS versions.

**Pull request merged**